tests/unit/test_security_middleware.py

```
@@ -0,0 +1,415 @@
+        assert response.status_code == 200
+
+    @pytest.mark.unit
+    def test_add_security_middleware_with_https_redirect(self) -> None:
+        """Verify add_security_middleware enables HTTPS redirect when requested."""
+        from fragrance_rater.middleware.security import add_security_middleware
+
+        app = FastAPI()
+        add_security_middleware(app, enable_https_redirect=True)
+
+        @app.get("/test")
+        async def _test_endpoint():
```

Ruff (0.15.12)

[warning] Missing return type annotation for private function `_test_endpoint`

(ANN202)